server/schemas/schemaMatcher.js

```diff
@@ -0,0 +1,80 @@
+var csv = require("csv");
+var Readable = require("stream").Readable;
+var schemaModel = require("./schemaModel");
+var helpers = require("../config/helpers");
+
+var handle = function(buffer, callback) {
+  var stringifier = new csv.stringify();
+  var parser = new csv.parse();
+  var readStream = new Readable({objectMode: true});
+  var validEntry = true;
+  var storage = {};
+  var columns;
+
+//turns an array into a query which can be sent to mongoDB
+//via mongoose in order to find a schema template with the same fields.
+  var queryBuilder = function(array) {
+    var obj = {};
```

Comment:
I still don't have a good idea what's going on here, but I'm not sure that `obj` needs to exist. It looks like a possibly unnecessary interface between `array` and `result`.